File: dogfood/snippets/C++/fizzbuzz.cpp
#include <iostream>

void fizzbuzz(int);

int main(void)
{
    for (int i = 1; i <= 100; i++) {
        fizzbuzz(i);
    }
}

void fizzbuzz(int num)
{
    if (num % 15 == 0) {
        std::cout << "FizzBuzz" << std::endl;
    } else if (num % 5 == 0) {
        std::cout << "Buzz" << std::endl;
    } else if (num % 3 == 0) {
        std::cout << "Fizz" << std::endl;
    } else {
        std::cout << num << std::endl;
    }
}

File: dogfood/snippets/Go/fizzbuzz.go
package main

import (
	"fmt"
)

func fizzbuzz(num int) {
	if num%3 == 0 && num%5 == 0 {
		fmt.Println("FizzBuzz")
	} else if num%5 == 0 {
		fmt.Println("Buzz")
	} else if num%3 == 0 {
		fmt.Println("Fizz")
	} else {
		fmt.Println(num)
	}
}

func main() {
	for i := 1; i < 100; i++ {
		fizzbuzz(i)
	}
}
